Feat/dashboard

minhnhat-vietqr opened the pull request.

minhnhat-vietqr committed:
merge

minhnhat-vietqr committed:
refactor

minhnhat-vietqr committed:
Merge branch 'development' into feat/dashboard

minhnhat-vietqr committed:
Add component & icon

minhnhat-vietqr committed:
refactor

minhnhat-vietqr committed:
refactor

minhnhat-vietqr committed:
Refactor redux store and add searchSlice

- Refactor the redux store to import the searchSlice and include it in the combineReducers function.
- Add the searchSlice file to the redux/slices directory.

minhnhat-vietqr committed:
Refactor localization and add new features

minhnhat-vietqr committed:
refactor

minhnhat-vietqr committed:
refactor

minhnhat-vietqr committed:
Refactor SearchBar component and add search functionality

minhnhat-vietqr committed:
rafactor

minhnhat-vietqr committed:
refactor

minhnhat-vietqr committed:
refactor horizontal dashboard

minhnhat-vietqr committed:
remove console log

minhnhat-vietqr committed:
refactor text gradient

minhnhat-vietqr committed:
refactor dashboard icons and add bill management feature

minhnhat-vietqr committed:
refactor

minhnhat-vietqr committed:
refactor

minhnhat-vietqr committed:
refactor

minhnhat-vietqr committed:
Merge branch 'development' into feat/dashboard

minhnhat-vietqr committed:
refactor

minhnhat-vietqr committed:
refactor

minhdt-vietqr committed:
improve animation

minhnhat-vietqr committed:
fix

minhnhat-vietqr committed:
fix

minhnhat-vietqr committed:
refactor

minhnhat-vietqr committed:
improve

minhnhat-vietqr committed:
improve

minhnhat-vietqr committed:
improve

minhnhat-vietqr committed:
improve

minhnhat-vietqr committed:
remove unused component

minhnhat-vietqr committed:
improve

minhnhat-vietqr committed:
improve

minhnhat-vietqr committed:
improve

minhnhat-vietqr committed:
add

minhnhat-vietqr committed:
improve

minhnhat-vietqr committed:
improve

minhnhat-vietqr committed:
add comment

minhnhat-vietqr committed:
Refactor HorizontalSidebar styles for better spacing and alignment

minhnhat-vietqr committed:
Refactor DashboardHeader and ExpandSearchBar components

minhnhat-vietqr committed:
Merge branch 'development' into feat/dashboard

minhnhat-vietqr committed:
Refactor HorizontalSidebar to set active parent tab based on current pathname

minhnhat-vietqr committed:
Refactor DashboardMode component to navigate to the first child of the selected dashboard type

minhnhat-vietqr committed:
Refactor Breadcrumbs component to remove console.log statement and handle optional breadcrumb properties

minhnhat-vietqr committed:
Refactor HorizontalSidebar to dispatch setDashboardType action on active parent tab change

minhnhat-vietqr committed:
Refactor ExpandSearchBar component to conditionally render search results and handle empty results

minhnhat-vietqr committed:
Refactor Breadcrumbs component to handle optional breadcrumb properties and use default icon if not provided

minhnhat-vietqr committed:
Refactor Breadcrumbs component to use a default icon if not provided

minhnhat-vietqr committed:
Refactor Breadcrumbs component to use DEFAULT_ICON if icon is not provided

minhnhat-vietqr committed:
Refactor DashboardMode component to navigate to the first child of the selected dashboard type

minhnhat-vietqr committed:
Refactor DashboardMode component to navigate to the first child of the selected dashboard type

minhnhat-vietqr committed:
Refactor menu.js to add default icons for menu items

minhnhat-vietqr committed:
remove unused

minhnhat-vietqr committed:
Add default icon for store management menu item

minhnhat-vietqr committed:
Add default icon for store management menu item

minhnhat-vietqr committed:
Refactor VerticalSidebar component to add default icons for menu items

minhnhat-vietqr committed:
Refactor DashboardMode component to add default icons for menu items

minhnhat-vietqr committed:
Add default icons for close and open menu items

minhnhat-vietqr committed:
Refactor VerticalSidebar component

minhnhat-vietqr committed:
Merge branch 'development' into feat/dashboard

minhnhat-vietqr committed:
Refactor VerticalSidebar component to use English labels for menu items

minhnhat-vietqr committed:
Refactor DashboardMode component to remove conditional borderRadius value

minhnhat-vietqr committed:
Refactor DashboardHeader and DashboardSidebar components to use constants for drawer width

minhnhat-vietqr committed:
Refactor DashboardHeader component to update backdrop filter blur value

minhnhat-vietqr committed:
Refactor AccountPopover component to wrap IconButton in a Box

minhnhat-vietqr committed:
Refactor Profile component to update styling and add responsive font size

minhnhat-vietqr committed:
Refactor VerticalSidebar component to update styling and add responsive layout

minhnhat-vietqr committed:
Refactor DashboardHeader component to update spacing values

minhnhat-vietqr committed:
Refactor DashboardHeader component to update top spacing value

minhnhat-vietqr committed:
Refactor DashboardHeader and VerticalSidebar components to update styling and add responsive layout

minhnhat-vietqr committed:
Refactor Breadcrumbs, LinkItem, CollapseSearchInput, ExpandSearchBar, VerticalSidebar, FeesApply, and FeesGrid components to update styling and use theme colors

minhnhat-vietqr committed:
Refactor DashboardHeader component to update spacing values

minhnhat-vietqr committed:
Refactor VerticalSidebar component to update spacing and padding values

minhnhat-vietqr committed:
Refactor VerticalSidebar component to update padding value

minhnhat-vietqr committed:
Refactor VerticalSidebar component to update padding and add cursor pointer

minhnhat-vietqr committed:
Refactor DashboardMode component to handle navigation to the first child of the selected dashboard type

minhnhat-vietqr committed:
Refactor DashboardMode component to navigate to the first child of the selected dashboard type

minhnhat-vietqr committed:
Refactor DashboardHeader and DashboardSidebar components to update layout and spacing values

minhnhat-vietqr committed:
Refactor DashboardMode component to disable ripple effect on navigation

minhnhat-vietqr committed:
Refactor localization files to add missing translations

minhnhat-vietqr committed:
Refactor Profile component to remove unused imports and update border style

minhnhat-vietqr committed:
Refactor AccountPopover component to update styling and remove unused imports

minhnhat-vietqr committed:
Refactor localization files to update "noNotification" translation

minhnhat-vietqr committed:
Refactor DashboardHeader component to update backdrop filter and box shadow styles

minhnhat-vietqr committed:
Refactor search bar and sidebar components

minhnhat-vietqr committed:
Refactor ExpandSearchBar component to disable autocomplete for search input

minhnhat-vietqr committed:
Merge branch 'development' into feat/dashboard

minhnhat-vietqr committed:
Refactor error.js to add client-side import statement

minhnhat-vietqr committed:
Refactor VerticalSidebar component to handle null values in displayedType

minhnhat-vietqr committed:
Refactor DashboardHeader, NotificationPopover and ExpandSearchBar components

minhnhat-vietqr committed:
Refactor VerticalSidebar component to center align and add padding to closed menu items

minhnhat-vietqr committed:
Refactor VerticalSidebar component to handle null values in displayedType

minhnhat-vietqr committed:
Refactor VerticalSidebar component to handle null values in displayedType and filter out "Settings" item

minhnhat-vietqr committed:
Refactor VerticalSidebar component to handle null values in displayedType and filter out "Settings" item

minhnhat-vietqr committed:
Refactor VerticalSidebar component to handle null values in displayedType and filter out "Settings" item

minhnhat-vietqr committed:
Refactor VerticalSidebar component to handle null values in displayedType and remove console.log statement

minhnhat-vietqr committed:
Refactor SearchBar and ExpandSearchBar components to handle search query clearing

minhnhat-vietqr committed:
Merge branch 'development' into feat/dashboard

minhnhat-vietqr committed:
Refactor VerticalSidebar component to handle null values in displayedType and remove console.log statement

minhnhat-vietqr committed:
Refactor VerticalSidebar component to simplify padding logic

minhnhat-vietqr committed:
Refactor VerticalSidebar component to simplify padding logic and handle null values in displayedType

minhnhat-vietqr committed:
Refactor SearchBar component to handle search query clearing

Refactor VerticalSidebar component to simplify padding logic and handle null values in displayedType

Refactor VerticalSidebar component to handle null values in displayedType and remove console.log statement

minhnhat-vietqr committed:
Refactor VerticalSidebar component to align items to the top instead of center

minhnhat-vietqr committed:
Refactor

minhnhat-vietqr committed:
Merge branch 'development' into feat/dashboard

minhnhat-vietqr committed:
Refactor SearchBar component to handle search query clearing

minhnhat-vietqr committed:
Refactor SearchBar component to handle search query clearing

minhnhat-vietqr committed:
Refactor SearchBar component to use default search result object

minhnhat-vietqr committed:
Refactor CustomDrawer component and add DrawerMobile component

minhnhat-vietqr committed:
Refactor DrawerMobile and DashboardHeader components to use onClose prop instead of setIsOpen and DashboardSidebar components to use onClick prop instead of setIsOpen

minhnhat-vietqr committed:
Refactor SearchBar component to use default search result object

minhnhat-vietqr committed:
Merge branch 'development' into feat/dashboard

minhnhat-vietqr committed:
Refactor Breadcrumbs component to use flexbox for layout

minhnhat-vietqr committed:
Refactor DashboardHeader component layout to use flexbox

minhnhat-vietqr committed:
Refactor change DASHBOARD_MODE from header to constants folder

minhdt-vietqr committed:
fix redux persist migrate

minhnhat-vietqr committed:
Refactor SearchBar component to adjust search container position based on dashboard mode

minhnhat-vietqr committed:
Refactor DashboardHeader component to remove console.log statement

minhnhat-vietqr committed:
Merge branch 'development' into feat/dashboard

minhnhat-vietqr committed:
Refactor VerticalSidebar component to add open state to ListItemButtonStyled

minhnhat-vietqr committed:
Refactor DrawerMobile component to import DASHBOARD_MODE from '~/constants/dashboard'

minhnhat-vietqr committed:
Refactor DashboardHeader component to import Z_INDEX from '~/constants/styles'

minhnhat-vietqr committed:
Refactor searchAPI to include searchTransaction function

minhdt-vietqr committed:
fix search bar

minhdt-vietqr committed:
remove wrapper

minhnhat-vietqr committed:
Merge branch 'feat/dashboard' of https://github.com/duchieupham/vietqr_web_2.0 into feat/dashboard

minhnhat-vietqr committed:
Refactor SearchBar component to remove unused imports and styles

minhnhat-vietqr committed:
Refactor searchService to include searchTransaction function

minhnhat-vietqr committed:
Refactor searchService to use numOfPastDays instead of daysAgo in searchTransaction function

minhnhat-vietqr committed:
Refactor searchService to use numOfPastDays instead of daysAgo in searchTransaction function

minhnhat-vietqr committed:
Refactor searchService to use numOfPastDays instead of daysAgo in searchTransaction function

minhnhat-vietqr committed:
Refactor useDebounce hook to add debouncedValue state and return it along with the current value

minhnhat-vietqr committed:
Refactor useDebounce hook to remove unused code and improve readability

minhnhat-vietqr committed:
Update dayjs to version 1.11.13

minhnhat-vietqr committed:
Refactor searchService to use dayjs for date calculations and update API endpoint in searchTransaction function

minhnhat-vietqr committed:
Refactor import paths in .eslintrc.json and add missing directories
Remove simplebar-react from package.json dependencies
Remove unused dependencies in yarn.lock

minhnhat-vietqr committed:
Merge branch 'development' into feat/dashboard

minhnhat-vietqr committed:
Refactor userID in searchTransaction function to use camel case

minhdt-vietqr commented on package.json:
```diff
@@ -45,7 +45,6 @@
     "redux-thunk": "^3.1.0",
     "sass": "^1.77.8",
     "sharp": "^0.33.5",
-    "simplebar-react": "^3.2.6",
```
cai nay hoi nao vay 🗡️

minhnhat-vietqr replied:
em không rõ mà em tìm thử trong project rồi ko có thằng nào sài hết nên êm remove luôn

minhdt-vietqr commented on package.json:
```diff
@@ -24,7 +24,7 @@
     "cookies-next": "^4.2.1",
     "crypto-js": "^4.2.0",
-    "dayjs": "^1.11.12",
+    "dayjs": "^1.11.13",
```
hạn chế thay đổi package json nếu k thật sự cần thiết

minhdt-vietqr commented on src/api/search/searchService.js:
```diff
@@ -1,16 +1,24 @@
+  const currentDate = dayjs().format('YYYY-MM-DD HH:mm:ss');
+  const numOfPastDays = dayjs()
+    .subtract(30, 'day')
+    .format('YYYY-MM-DD HH:mm:ss');
```
sao phải format 2 lần?

minhnhat-vietqr replied:
tại em lấy từ thằng default của dayjs rồi tìm lại 30 ngày trước, chứ không lấy thằng ```currentDate``` trừ lại 30 anh ơi, nên em mới format lại 2 lần

minhnhat-vietqr replied:
với một phần khi em lấy thằng ```currentDate``` để subtract thì nó không nhận được hàm nên em mới tách ra luôn
![{95262C8A-44AF-4315-B7B4-E03F7EA720AF}](https://github.com/user-attachments/assets/1417167c-41bd-4004-bef1-b9855a20c9ab)

minhdt-vietqr replied:
vì đi format trước rồi trừ thì bị là đúng rồi

minhdt-vietqr replied:
xử lí hết tới cuối mới format chứ

minhnhat-vietqr replied:
![{41DF31EE-B374-4272-9D41-0A0A0F805124}](https://github.com/user-attachments/assets/198bb107-658e-4d6e-8dbe-0f70689de338)
mình như thế này hả anh

minhdt-vietqr replied:
Ok

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -153,30 +155,38 @@ const DEFAULT_SEARCH_RESULT = {
       item.children.forEach((child) => {
-        const labelConverted = t(child.label);
+        const labelConverted = t(child.label); // Convert label to text for searching in translation
         if (labelConverted.toLowerCase().includes(query.toLowerCase())) {
```
check null vì translate có thể ra

minhdt-vietqr replied:
nên trim query

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -202,6 +214,12 @@ export default function SearchBar() {
   const [isExpanded, setIsExpanded] = useState(false);
   const [searchQuery, setSearchQuery] = useState('');
+  const [searchResult, setSearchResult] = useState({
+    feat: [],
```
dung constant

minhnhat-vietqr committed:
Refactor searchByLabel function to skip empty labels in search results

minhnhat-vietqr committed:
Refactor SearchBar component to use default search result constant

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -210,24 +228,72 @@ export default function SearchBar() {
-      setSearchResult(result);
+
+      // check if all categories are empty
+      if (Object.values(result).every((category) => category.length === 0)) {
```
lí do cho sự cồng kềnh này vì default có sẵn 4 cái empty array. Nếu default là null thì clear hơn và gọn hơn r

minhdt-vietqr replied:
tương tự ở chỗ render

minhnhat166 replied:
chỗ dùng useMemo hả anh

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -210,24 +228,72 @@ export default function SearchBar() {
-  const onSearchChange = (e) => {
-    const query = e.target.value;
-    setSearchQuery(query);
+  const onSearchChange = async (query) => {
```
name lại là handleSearch vì đổi cả chỗ xài

minhdt-vietqr replied:
set result quá nhiều lúc xử lí, cần làm là từ query xử lí hết rồi mới cho ra result, này kiểu check 1 đứa r set, xong lại check tiếp. Flow vâỵ rất ba chấm kiểu làm k tới nơi đã trả về

minhnhat-vietqr committed:
Refactor searchTransaction function to use camel case for userId and update date formatting

minhnhat-vietqr committed:
Refactor searchByLabel function to handle empty labels in search results

minhdt-vietqr commented on src/api/search/searchService.js:
```diff
@@ -1,16 +1,23 @@
-  const currentDate = new Date();
-  const numOfPastDays = currentDate.getDate() - 30; // 30 days ago
-  const param = {
+export const DATE_REGEX = 'YYYY-MM-DD HH:mm:ss';
```
cái này nếu format cho dùng đa số api thì mang qua bên global constant cho dễ quản lí. Ko thì khỏi export vì ở đây k define component hay service gì để expose cho reuse chỗ khác

minhnhat-vietqr committed:
Refactor searchService.js and index.js

- Update searchTransaction function in searchService.js to use camel case for userId and update date formatting
- Add DATE_REGEX constant to index.js for date formatting

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -153,30 +155,51 @@ const DEFAULT_SEARCH_RESULT = {
-        const labelConverted = t(child.label);
-        if (labelConverted.toLowerCase().includes(query.toLowerCase())) {
-          searchResult[item.label].push(child);
+        const labelConverted = t(child.label).trim().toLowerCase(); // Convert label to text for searching in translation
```
trim ở đây r thì ở dưới k cần nữa. Với lại check null ngắn gọn bằng optional chaining đi vì k có xử lí phức tạp chia case

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -153,30 +155,51 @@ const DEFAULT_SEARCH_RESULT = {
-        if (labelConverted.toLowerCase().includes(query.toLowerCase())) {
-          searchResult[item.label].push(child);
+        const labelConverted = t(child.label).trim().toLowerCase(); // Convert label to text for searching in translation
+        if (!labelConverted.includes(query)) {
```
trim là phải trim cái query k kiếm soát đc đây

minhnhat-vietqr replied:
query khuc nay la em truyen vao o thang handleSearch la co trim() roi a anh

minhnhat-vietqr committed:
Refactor searchByLabel function to skip empty labels in search results

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -202,38 +227,73 @@ export default function SearchBar() {
+    const result = searchByLabel(lowerCaseQuery, t);
+
+    // search transaction if query length >= 10 characters
+    if (lowerCaseQuery.length >= 10) {
```
requirement à?

minhnhat-vietqr replied:
anh Linh ảnh sợ call nhiều nó lag server nên ổng bảo thêm vào á anh

minhdt-vietqr replied:
transaction có id theo pattern gì ko thì check regex mới call, hiện tại là dùng debounce để tránh gọi liên tục r, check length >= 10 cũng đc th nhưng hơi hard

minhnhat-vietqr replied:
[
    {
        "transactionId": "2bbef738-b4d3-4a8a-811d-b3384a9fc1cf",
        "amount": "1,000,000",
        "content": "NGUYEN PHUONG NHAI LINH chuyen tien- Ma GD ACSP/ 2W799448",
        "qrCode": "",
        "time": 1728874830,
        "timePaid": 1728874830,
        "status": 1,
        "type": 2,
        "transType": "D",
        "referenceNumber": "FT24288185696734",
        "orderId": "",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=MmJiZWY3MzgtYjRkMy00YThhLTgxMWQtYjMzODRhOWZjMWNm",
        "terminalCode": ""
    },
    {
        "transactionId": "0acb215a-9603-4f2f-b598-9fc22ba3acf6",
        "amount": "1,000,000",
        "content": "PHAN NHAT MINH chuyen tien FT24288101087205   Ma giao dich  Trace179716 Trace 179716",
        "qrCode": "",
        "time": 1728874722,
        "timePaid": 1728874722,
        "status": 1,
        "type": 2,
        "transType": "C",
        "referenceNumber": "FT24288381074754",
        "orderId": "",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=MGFjYjIxNWEtOTYwMy00ZjJmLWI1OTgtOWZjMjJiYTNhY2Y2",
        "terminalCode": ""
    },
    {
        "transactionId": "dfa7ed8f-7839-438f-a1b4-c3bbd1a3989c",
        "amount": "5,000",
        "content": "BLC60 Linhsky01236 Refund",
        "qrCode": "",
        "time": 1728633088,
        "timePaid": 1728633088,
        "status": 1,
        "type": 0,
        "transType": "D",
        "referenceNumber": "FT24285561002512",
        "orderId": "Linhsky01236",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=ZGZhN2VkOGYtNzgzOS00MzhmLWExYjQtYzNiYmQxYTM5ODlj",
        "terminalCode": ""
    },
    {
        "transactionId": "c4406c5d-af42-4ff9-9f97-2b690162d742",
        "amount": "5,000",
        "content": "Test",
        "qrCode": "00020101021238570010A000000727012700069704220113VQRQAASGH48660208QRIBFTTA5303704540450005802VN62190107NPS68690804Test630456FA",
        "time": 1728632911,
        "timePaid": 1728632974,
        "status": 1,
        "type": 0,
        "transType": "C",
        "referenceNumber": "FT24285905944265",
        "orderId": "Linhsky01236",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=YzQ0MDZjNWQtYWY0Mi00ZmY5LTlmOTctMmI2OTAxNjJkNzQy",
        "terminalCode": ""
    },
    {
        "transactionId": "7bb135f7-0b20-4b30-b7bb-250af5b93331",
        "amount": "5,000",
        "content": "Test",
        "qrCode": "00020101021238570010A000000727012700069704220113VQRQAASGH06700208QRIBFTTA5303704540450005802VN62190107NPS68690804Test63045664",
        "time": 1728632424,
        "timePaid": 1728632506,
        "status": 1,
        "type": 0,
        "transType": "C",
        "referenceNumber": "FT24285197645335",
        "orderId": "LINHSKY106",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=N2JiMTM1ZjctMGIyMC00YjMwLWI3YmItMjUwYWY1YjkzMzMx",
        "terminalCode": ""
    },
    {
        "transactionId": "b7d26e08-7dfd-4703-b2a6-b4cda6d52cd2",
        "amount": "5,000",
        "content": "Test",
        "qrCode": "00020101021238570010A000000727012700069704220113VQRQAASGH08120208QRIBFTTA5303704540450005802VN62190107NPS68690804Test63041A05",
        "time": 1728632413,
        "timePaid": 1728632488,
        "status": 1,
        "type": 0,
        "transType": "C",
        "referenceNumber": "FT24285036382392",
        "orderId": "Linhsky01235",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=YjdkMjZlMDgtN2RmZC00NzAzLWIyYTYtYjRjZGE2ZDUyY2Qy",
        "terminalCode": ""
    },
    {
        "transactionId": "3330762e-5301-4533-870f-19a38354be74",
        "amount": "5,000",
        "content": "BLC60 Linhsky01231 Refund",
        "qrCode": "",
        "time": 1728632363,
        "timePaid": 1728632363,
        "status": 1,
        "type": 0,
        "transType": "D",
        "referenceNumber": "FT24285898001900",
        "orderId": "Linhsky01231",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=MzMzMDc2MmUtNTMwMS00NTMzLTg3MGYtMTlhMzgzNTRiZTc0",
        "terminalCode": ""
    },
    {
        "transactionId": "86c2c4ea-0fe6-4822-82e7-b2d789e9df51",
        "amount": "5,000",
        "content": "BLC60 LINHSKY101 Refund - Ma giao dich/ Trace 341049",
        "qrCode": "",
        "time": 1728632363,
        "timePaid": 1728632363,
        "status": 1,
        "type": 0,
        "transType": "D",
        "referenceNumber": "FT24285769303108",
        "orderId": "LINHSKY101",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=ODZjMmM0ZWEtMGZlNi00ODIyLTgyZTctYjJkNzg5ZTlkZjUx",
        "terminalCode": ""
    },
    {
        "transactionId": "0fcbf3ea-cf2d-43e8-a873-807b223efd0c",
        "amount": "5,000",
        "content": "BLC60 LINHSKY102 Refund - Ma giao dich/ Trace 335441",
        "qrCode": "",
        "time": 1728632302,
        "timePaid": 1728632302,
        "status": 1,
        "type": 0,
        "transType": "D",
        "referenceNumber": "FT24285502701901",
        "orderId": "LINHSKY102",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=MGZjYmYzZWEtY2YyZC00M2U4LWE4NzMtODA3YjIyM2VmZDBj",
        "terminalCode": ""
    },
    {
        "transactionId": "5fa0dbae-2c33-4985-a6f4-d0fac4f17e6a",
        "amount": "5,000",
        "content": "BLC60 Linhsky01232 Refund",
        "qrCode": "",
        "time": 1728632301,
        "timePaid": 1728632301,
        "status": 1,
        "type": 0,
        "transType": "D",
        "referenceNumber": "FT24285003574351",
        "orderId": "Linhsky01232",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=NWZhMGRiYWUtMmMzMy00OTg1LWE2ZjQtZDBmYWM0ZjE3ZTZh",
        "terminalCode": ""
    },
    {
        "transactionId": "a211c0f6-0172-4f54-8d8d-bd1fab06cd24",
        "amount": "5,000",
        "content": "BLC60 LINHSKY103 Refund - Ma giao dich/ Trace 321041",
        "qrCode": "",
        "time": 1728632127,
        "timePaid": 1728632127,
        "status": 1,
        "type": 0,
        "transType": "D",
        "referenceNumber": "FT24285350862437",
        "orderId": "LINHSKY103",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=YTIxMWMwZjYtMDE3Mi00ZjU0LThkOGQtYmQxZmFiMDZjZDI0",
        "terminalCode": ""
    },
    {
        "transactionId": "c44fec39-7240-4d4c-9c29-f399b5933977",
        "amount": "5,000",
        "content": "BLC60 Linhsky01233 Refund",
        "qrCode": "",
        "time": 1728632126,
        "timePaid": 1728632126,
        "status": 1,
        "type": 0,
        "transType": "D",
        "referenceNumber": "FT24285538113450",
        "orderId": "Linhsky01233",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=YzQ0ZmVjMzktNzI0MC00ZDRjLTljMjktZjM5OWI1OTMzOTc3",
        "terminalCode": ""
    },
    {
        "transactionId": "90d3ecd9-220d-4976-a435-18cade4af39c",
        "amount": "5,000",
        "content": "BLC60 Linhsky01230 Refund",
        "qrCode": "",
        "time": 1728632035,
        "timePaid": 1728632035,
        "status": 1,
        "type": 0,
        "transType": "D",
        "referenceNumber": "FT24285080302426",
        "orderId": "Linhsky01230",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=OTBkM2VjZDktMjIwZC00OTc2LWE0MzUtMThjYWRlNGFmMzlj",
        "terminalCode": ""
    },
    {
        "transactionId": "bd3bfcd9-1f8c-4ae4-afa6-ded12556dd15",
        "amount": "5,000",
        "content": "BLC60 Linhsky01234 Refund",
        "qrCode": "",
        "time": 1728632035,
        "timePaid": 1728632035,
        "status": 1,
        "type": 0,
        "transType": "D",
        "referenceNumber": "FT24285767006807",
        "orderId": "Linhsky01234",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=YmQzYmZjZDktMWY4Yy00YWU0LWFmYTYtZGVkMTI1NTZkZDE1",
        "terminalCode": ""
    },
    {
        "transactionId": "0b6cafac-527e-4ecf-a412-3f72e4a8169e",
        "amount": "5,000",
        "content": "BLC60 LINHSKY105 Refund - Ma giao dich/ Trace 081505",
        "qrCode": "",
        "time": 1728631955,
        "timePaid": 1728631955,
        "status": 1,
        "type": 0,
        "transType": "D",
        "referenceNumber": "FT24285489170412",
        "orderId": "LINHSKY105",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=MGI2Y2FmYWMtNTI3ZS00ZWNmLWE0MTItM2Y3MmU0YTgxNjll",
        "terminalCode": ""
    },
    {
        "transactionId": "03dde760-0863-48a7-9987-9963f326e63f",
        "amount": "5,000",
        "content": "BLC60 LINHSKY104 Refund - Ma giao dich/ Trace 309239",
        "qrCode": "",
        "time": 1728631951,
        "timePaid": 1728631951,
        "status": 1,
        "type": 0,
        "transType": "D",
        "referenceNumber": "FT24285097785657",
        "orderId": "LINHSKY104",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=MDNkZGU3NjAtMDg2My00OGE3LTk5ODctOTk2M2YzMjZlNjNm",
        "terminalCode": ""
    },
    {
        "transactionId": "920d9812-c412-4036-b619-5160328ae063",
        "amount": "5,000",
        "content": "Test",
        "qrCode": "00020101021238570010A000000727012700069704220113VQRQAASFW26120208QRIBFTTA5303704540450005802VN62190107NPS68690804Test630478E3",
        "time": 1728618682,
        "timePaid": 1728618962,
        "status": 1,
        "type": 0,
        "transType": "C",
        "referenceNumber": "FT24285681063772",
        "orderId": "LINHSKY105",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=OTIwZDk4MTItYzQxMi00MDM2LWI2MTktNTE2MDMyOGFlMDYz",
        "terminalCode": ""
    },
    {
        "transactionId": "8420dfd5-1b3b-4d61-a012-7661cbc65cd5",
        "amount": "5,000",
        "content": "Test",
        "qrCode": "00020101021238570010A000000727012700069704220113VQRQAASFV99280208QRIBFTTA5303704540450005802VN62190107NPS68690804Test6304D603",
        "time": 1728618194,
        "timePaid": 1728618816,
        "status": 1,
        "type": 0,
        "transType": "C",
        "referenceNumber": "FT24285947306435",
        "orderId": "Linhsky01234",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=ODQyMGRmZDUtMWIzYi00ZDYxLWEwMTItNzY2MWNiYzY1Y2Q1",
        "terminalCode": ""
    },
    {
        "transactionId": "6a80d35e-0a22-4188-a8b8-bf3b5b504825",
        "amount": "5,000",
        "content": "Test",
        "qrCode": "00020101021238570010A000000727012700069704220113VQRQAASFV98920208QRIBFTTA5303704540450005802VN62190107NPS68690804Test6304F9BA",
        "time": 1728618184,
        "timePaid": 1728618779,
        "status": 1,
        "type": 0,
        "transType": "C",
        "referenceNumber": "FT24285103860433",
        "orderId": "Linhsky01233",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=NmE4MGQzNWUtMGEyMi00MTg4LWE4YjgtYmYzYjViNTA0ODI1",
        "terminalCode": ""
    },
    {
        "transactionId": "ba1a37ec-d3f3-4009-b0a0-ae1dcb5ffbd5",
        "amount": "5,000",
        "content": "Test",
        "qrCode": "00020101021238570010A000000727012700069704220113VQRQAASFV98460208QRIBFTTA5303704540450005802VN62190107NPS68690804Test6304E555",
        "time": 1728618176,
        "timePaid": 1728618736,
        "status": 1,
        "type": 0,
        "transType": "C",
        "referenceNumber": "FT24285040097745",
        "orderId": "Linhsky01232",
        "bankAccount": "0373568944",
        "userBankName": "Nguyen Phuong Nhai Linh",
        "bankCode": "MB",
        "bankName": "Ngân hàng TMCP Quân đội",
        "imgId": "58b7190b-a294-4b14-968f-cd365593893e",
        "bankShortName": "MBBank",
        "qrLink": "https://vietqr.vn/qr-generated?token=YmExYTM3ZWMtZDNmMy00MDA5LWIwYTAtYWUxZGNiNWZmYmQ1",
        "terminalCode": ""
    }
]

có anh ơi mà em chưa thấy requirement về việc hiển thị để cái gì lên nên code đang đại anh ơi,

minhnhat-vietqr replied:
với lại 1 phần là kiểu mình nhập 2 kí tự thì nó đã gọi rồi mà server lỏ quá nên ổng run anh ơi, qua mới sập server

minhdt-vietqr replied:
vây cũng đc th, này do requirment chung chung search cả đám thì chấp nhận
nếu muốn đỡ thì cho nhập thêm pattern / có thể thêm phần select để chọn search cái gì thì ổn hơn chứ tần suất gọi api thì cũng k bùm nhiều mà liên tục đc

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -202,38 +227,73 @@ export default function SearchBar() {
+        // userId: '648dca06-4f72-4df8-b98f-429f4777fbda', // test
+      });
+
+      if (transactionResult?.data?.transactions?.length) {
```
ai lại đi check null length, check api có data hoặc ko thôi

minhnhat166 replied:
dạ okee anh

minhnhat-vietqr committed:
Refactor SearchBar component to update transaction result handling

minhdt-vietqr commented on src/api/search/searchService.js:
```diff
@@ -1,16 +1,22 @@
+    useAuth: true,
+    params: params,
+  });
   return res;
```
dùng axios thì nó trả về 4 object, default thì check data trả về luôn hoặc trả null thì lúc dùng k cần chấm cũng như có mấy cái k dùng tới

minhnhat-vietqr replied:
![{F434F7B7-9D29-4193-B376-7319880EF0AF}](https://github.com/user-attachments/assets/edca3190-edee-48f6-9a94-028bab2534e6)

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -202,38 +221,73 @@ export default function SearchBar() {
+      });
+
+      if (transactionResult?.data) {
+        result.transaction = transactionResult.data.transactions.map((txn) => ({
```
k có check null transaction, map ra bug

minhnhat-vietqr committed:
Refactor searchService.js to handle empty search results

minhnhat-vietqr committed:
Refactor SearchBar component to handle loading state during search

minhnhat-vietqr committed:
Refactor SearchBar component to handle loading state during search
Not yet handle LoadingContainer

minhnhat-vietqr committed:
Refactor SearchBar component to handle loading state during search

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -153,46 +157,54 @@ const DEFAULT_SEARCH_RESULT = {
+          <Box>{t(item.label)}</Box>
+        </ListItem>
+      )),
+    [result, t],
```
`t` ko có đổi để mà bỏ vô deps của memo đâu

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -384,9 +438,43 @@ export default function SearchBar() {
+                <Box sx={{ display: 'flex', justifyContent: 'center' }}>
+                  <GradientCircularProgress />
+                </Box>
+              ) : ITEMS.some((item) => searchResult[item.label].length > 0) ? (
```
check khá cồng kềnh, trọng tâm là `searchResult` lôi `ITEMS` vô, viết 1 cái memo func để check `searchResult` có empty hay ko.
1. duyệt từng item
2. check length
có 1 cái > 0 return true, ko thì mặc định return false

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -384,9 +438,43 @@ export default function SearchBar() {
+                  <GradientCircularProgress />
+                </Box>
+              ) : ITEMS.some((item) => searchResult[item.label].length > 0) ? (
+                ITEMS.map((item) => {
```
sau đó render dựa theo `searchResult `, map thì từ `ITEMS` xong rồi lại dùng result check nó rườm rà, vòng vo trong khi cái quan trọng là xoay quanh `searchResult`
ITEMS chỉ là data gốc, lọc ra bỏ vô `searchResult` rồi. Cấu trúc 2 cái cũng đang y chang

minhdt-vietqr commented on src/components/feedback/loading/LoadingWithLogo.js:
```diff
@@ -1,7 +1,7 @@
 import Image from 'next/image';
 
-const GradientCircularProgress = () => (
+export const GradientCircularProgress = () => (
```
đang dùng cho riêng component LoadingWithLogo này.
muốn reuse thì tách ra 1 file chứ component này k có nhiệm vụ export ra 1 cái component khác ko liên quan ngoài nó

minhnhat-vietqr committed:
Refactor

minhdt-vietqr commented on src/utils/getImage.js:
```diff
@@ -1,6 +1,6 @@
 
-const useImage = (imageId) => {
+const getImage = (imageId) => {
   const [imageUri, setImageUri] = useState('');
```
bỏ state, k phải hook nữa, chỉ trả về string

minhdt-vietqr commented on src/components/Profile.js:
```diff
@@ -1,12 +1,12 @@
 export default function Profile({ ...props }) {
   const { session } = useAuthContext();
-  const imageUrl = useImage(session?.imgId);
+  const imageUrl = getImage(session?.imgId);
```
có thể memo

minhnhat-vietqr committed:
Refactor constants and utils to use BASE_URL from environment variables

minhnhat-vietqr committed:
Refactor Profile component to use useMemo for image URL

minhnhat-vietqr committed:
Refactor Profile component to use useMemo for image URL

minhnhat-vietqr committed:
Refactor SearchBar component to remove console.log statement

minhdt-vietqr commented on src/constants/index.js:
```diff
@@ -44,3 +44,9 @@ export const LANGUAGE_OPTIONS = [
+export const DATE_REGEX = 'YYYY-MM-DD HH:mm:ss';
+
+// API
+export const BASE_URL = process.env.NEXT_PUBLIC_API_URL;
```
file mới, config.js

minhdt-vietqr commented on src/sections/login/navbar/NavbarLogin.js:
```diff
@@ -5,23 +5,26 @@ import { Box, Button, Drawer, Grid, IconButton, Stack } from '@mui/material';
-  const imageUri = useImage(AppImages.logoVietQr);
+  const imageUri = useMemo(
+    () => getImage(AppImages?.logoVietQr),
+    [AppImages?.logoVietQr],
```
ở đây constant k đổi nên k cần deps

minhnhat-vietqr committed:
Refactor constants to use BASE_URL from environment variables

minhdt-vietqr commented on src/utils/getImage.js:
```diff
@@ -0,0 +1,9 @@
+const getImage = (imageId) => {
+  if (!imageId) return null; // Return early if imageId is not provided.
+  const stringUrl = `${BASE_URL}/images/${imageId}`;
+  return stringUrl;
```
có thể gọn lại bằng 3 ngôi `imageId ?  `${BASE_URL}/images/${imageId}` : null`

minhnhat-vietqr committed:
Refactor NavbarLogin component to use useMemo for image URL

minhnhat-vietqr committed:
Refactor getImage function to handle null imageId

minhdt-vietqr commented on src/components/SearchBar.js:
```diff
@@ -202,40 +214,87 @@ export default function SearchBar() {
-      setSearchResult(DEFAULT_SEARCH_RESULT);
+  const handleSearch = async (query) => {
+    setIsLoading(true);
+    const lowerCaseQuery = query.trim().toLowerCase();
```
`searchQuery` hoặc `parsedQuery` hoặc `searchKeyword` thoi, name kiểu này cũng k đủ nghĩa vì có cả trim

minhnhat-vietqr committed:
Refactor SearchBar component to use parsedQuery instead of lowerCaseQuery in handleSearch function

minhnhat-vietqr committed:
Merge branch 'development' into feat/dashboard
